### PR TITLE
【Hackathon No.68】decouple GetDataFromTensor for phi

### PR DESCRIPTION
**Pull request by jinyouzhi**
### PR types

Others

### PR changes

Others

### Describe

There are 4 funcs in `operators/utils.h`:
- `GetDataFromTensor`: Only `GetDataFromTensor` function is used as deps in phi. 
- `GetDataFromTensorList`
- `GetShape`
- `GetValue`: Phi has native `GetValue`.

So just seperate `GetDataFromTensor` from fluid to phi, I place it around `GetValue` in phi/core/tensor_utils.h.

https://github.com/PaddlePaddle/Paddle/issues/50659#task68

### CODE REVIEW

**Comment by jinyouzhi** [comment]
Since the current code has passed most CI except the following 2, Could anyone start the review? @luotao1 @YuanRisheng @cloud2009 Thanks!
 - Coverage (line coverage <90%)
 - OP-Benchmarking ([strange type error](https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/7853872/job/21829034) for GPU Time output 0 both PR and develop)

File: paddle/fluid/operators/utils.h [hunk 1]
```diff
@@ -14,45 +14,14 @@ limitations under the License. */
 
 #pragma once
 #include <paddle/fluid/framework/operator.h>
+#include <paddle/phi/core/tensor_utils.h>
```
Comment (YuanRisheng) [inline paddle/fluid/operators/utils.h]: 这里建议改成用引号使用头文件

File: paddle/phi/core/tensor_utils.h [hunk 2]
```diff
@@ -144,4 +147,36 @@ inline T GetValue(const Context& dev_ctx, const DenseTensor& x) {
   return value;
 }
 
+template <typename T = int32_t>
+inline std::vector<T> GetDataFromTensor(const DenseTensor* x) {
```
Comment (YuanRisheng) [inline paddle/phi/core/tensor_utils.h]: 这个改个名字吧，叫GetVectorFromTensor或者TensorToVector

File: paddle/fluid/operators/utils.h (hunk 1 above)
Comment (jinyouzhi) [reply paddle/fluid/operators/utils.h]: > 这里建议改成用引号使用头文件

已更改

File: paddle/phi/core/tensor_utils.h (hunk 2 above)
Comment (jinyouzhi) [reply paddle/phi/core/tensor_utils.h]: > 这个改个名字吧，叫GetVectorFromTensor或者TensorToVector

已更改为GetVectorFromTensor

**Comment by YuanRisheng** [comment]
你好，这个题目已经有另一位开发者在你之前完成了PR review及合入，你稍微晚了一步

**Pull request closed without merging**